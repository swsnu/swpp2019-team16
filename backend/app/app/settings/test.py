from .base import *
import environ
from redis import ConnectionPool

env = environ.Env()
env.read_env('env/.env.development')

DEBUG = True

# Database
# https://docs.djangoproject.com/en/2.1/ref/settings/#databases

DATABASES = {
    'default': {
        'ENGINE': 'django.db.backends.sqlite3',
        'NAME': os.path.join(BASE_DIR, 'db.sqlite3'),
    }
}

REDIS_CONNECTION_POOL = ConnectionPool(host=env('REDIS_HOST'), port=6379, db=0)
