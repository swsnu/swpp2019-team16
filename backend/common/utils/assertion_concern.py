class AssertionConcern:

    def assert_not_none(self, val, msg):
        if val is None:
            raise TypeError(msg)
