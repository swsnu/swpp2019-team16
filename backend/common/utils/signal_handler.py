import asyncio
import functools
import signal


def register_signal_handler(loop, shutdown):
    for signame in ('SIGINT', 'SIGTERM'):
        loop.add_signal_handler(
            getattr(signal, signame),
            functools.partial(asyncio.ensure_future,
                              shutdown(signame, loop)))


async def shutdown_process(self, sig, loop):
    print('caught {0}'.format(sig))
    tasks = [task for task in asyncio.Task.all_tasks() if task is not
             asyncio.tasks.Task.current_task()]

    list(map(lambda task: task.cancel(), tasks))
    results = await asyncio.gather(*tasks, return_exceptions=True)

    print('finished awaiting cancelled tasks, results: {0}'
          .format(results))
    loop.stop()
