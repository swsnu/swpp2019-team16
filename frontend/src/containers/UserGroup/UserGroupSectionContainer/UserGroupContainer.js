import React, { useCallback, useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import UserGroupSection from '../../../components/UserGroup/UserGroupSection';
import { withRouter, useHistory } from 'react-router-dom';
import * as grpcClient from '../../../lib/grpc/client';
import {
  DRIVER_GO_TAXI_EVENT,
  RIDER_ON_TAXI_EVENT,
} from '../../../types/event';
import {
  taxiGone,
  goTaxi,
  riderOnTaxi,
  someoneOnTaxi,
} from '../../../modules/user';

function UserGroupContainer() {
  const history = useHistory();
  const dispatch = useDispatch();

  const { user, group, onTaxiRidersList, driverGoTaxi } = useSelector(
    ({ user, group }) => ({
      user: user.user,
      group: group.group,
      onTaxiRidersList: user.onTaxiRidersList,
      driverGoTaxi: user.driverGoTaxi,
    }),
  );

  useEffect(() => {
    if (user) {
      const stream = grpcClient.createGrpcStream({ id: user.user.id });
      stream.on('data', message => {
        const parsed = JSON.parse(message.getData());
        if (parsed._type_name === RIDER_ON_TAXI_EVENT) {
          dispatch(
            someoneOnTaxi({
              riderId: parsed._rider.id,
            }),
          );
        }

        if (parsed._type_name === DRIVER_GO_TAXI_EVENT) {
          dispatch(taxiGone());
        }
      });
      return () => {
        stream.cancel();
      };
    }
  }, [user]);

  useEffect(() => {
    if (driverGoTaxi) {
      if (user.user.user_type === 'rider') {
        history.push('/riderdetail');
      } else {
        history.push('/driverdetail');
      }
    }
  }, [driverGoTaxi, user]);

  const onClickOnTaxi = useCallback(() => {
    dispatch(riderOnTaxi({ riderId: user.id }));
  }, [dispatch, user]);

  const onClickGoTaxi = useCallback(() => {
    dispatch(goTaxi({ driverId: user.id }));
  }, [dispatch, user]);

  if (!group) {
    return <div>We are loading your group...</div>;
  }

  return (
    <UserGroupSection
      user={user}
      group={group}
      onTaxiRidersList={onTaxiRidersList}
      onClickOnTaxi={onClickOnTaxi}
      onClickGoTaxi={onClickGoTaxi}
    />
  );
}
export default withRouter(UserGroupContainer);
