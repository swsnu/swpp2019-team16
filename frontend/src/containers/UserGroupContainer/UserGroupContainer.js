import React, { useCallback } from 'react';
import { useDispatch } from 'react-redux';
import UserGroup from '../../components/UserGroup/UserGroup';
import { withRouter } from 'react-router-dom';
import { onTaxi } from '../../modules/group/group';

// todo: need to send Google Map API as props
// todo: need to add timer
// todo: need to get data from backend
// current state : group has been formed

function UserGroupContainer() {
  const dispatch = useDispatch();
  const onClickOnTaxi = useCallback(({ id }) => dispatch(onTaxi(id)), [
    dispatch,
  ]);
  const googleMap = {
    map: 'Google Map',
  };
  const driverInfo = {
    Name: 'MockDriver',
    Vehicle: 'BMW',
    Plate: '01A 1234',
  };
  const group = ['Rider1', 'Rider2', 'Rider3', 'Rider4'];
  return (
    <UserGroup
      group={group}
      onClick={onClickOnTaxi}
      googleMap={googleMap}
      driverInfo={driverInfo}
    />
  );
}
export default withRouter(UserGroupContainer);
