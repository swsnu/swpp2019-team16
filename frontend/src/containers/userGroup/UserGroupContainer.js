import React, { useEffect, useCallback } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import UserGroup from '../../components/userGroup/UserGroup';
import { withRouter } from 'react-router-dom';


// todo: need to send Google Map API as props
// todo: need to send driver info as props


function UserGroupContainer() {
    // const dispatch = useDispatch();
    // const driverInfo = useSelector( state => state.driverInfo );
    return (
        <UserGroup />
    );
}
export default withRouter(UserGroupContainer);