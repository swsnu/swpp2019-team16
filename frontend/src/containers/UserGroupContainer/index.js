import UserGroupContainer from './UserGroupContainer';
export default UserGroupContainer;
