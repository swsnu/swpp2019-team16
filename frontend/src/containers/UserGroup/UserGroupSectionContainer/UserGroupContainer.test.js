import React from 'react';
import UserGroupContainer from './UserGroupContainer';
import { renderWithRedux } from '../../../test/utils';
import { MemoryRouter, Route } from 'react-router-dom';
import expectExport from 'expect';

jest.mock('../../../components/UserGroup/UserGroupSection', () =>
  jest.fn(props => <div></div>),
);

describe('<UserGroupSectionContainer />', () => {
  it('SHOULD match with snapshot', async () => {
    const { container } = renderWithRedux(
      <MemoryRouter initialEntries={['/group']}>
        <Route component={UserGroupContainer} path="/group" />
      </MemoryRouter>,
    );
    expectExport(container).toMatchSnapshot();
  });
});
