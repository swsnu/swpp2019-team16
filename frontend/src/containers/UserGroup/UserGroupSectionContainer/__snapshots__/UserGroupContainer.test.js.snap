// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserGroupSectionContainer /> SHOULD match with snapshot 1`] = `
<div>
  <div />
</div>
`;
